### PR TITLE
31/clear chat

### LINKED ISSUE #31: Delete all messages in chat
Whipe out every message in the chat

### PR DESCRIPTION
**Pull request by spaenleh**
This PR adds admin tools to clear chat (displayed to admins).

Admin tools:
Screenshot 2022-04-19 at 17 20 51

Confirm dialog:
Screenshot 2022-04-19 at 17 20 40

closes #31

### CODE REVIEW

File: src/test/admin-tools.test.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      .should('exist')
+      .and('be.visible')
+      // download file
+      .click();
+
+    // get file name from data-cy-filename attribute and check local csv
+    cy.get(dataCyWrapper(exportChatButtonCypress))
+      .should('have.attr', 'data-cy-filename')
+      .then((filename) => {
+        console.log('filename:', filename);
```
Comment (pyphilia) [inline src/test/admin-tools.test.tsx:61]: remove

File: src/test/admin-tools.test.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+  it('should not show export chat with empty chat', () => {
+    // show tools but with no data
+    mountChatbox(true, true);
+    cy.get(dataCyWrapper(adminToolsContainerCypress)).should('exist');
+    cy.get(dataCyWrapper(exportChatButtonCypress)).should('not.exist');
+  });
+
+  it('should show clear chat button', () => {
+    const clearChatSpy = spyMethod('spyMethod');
```
Comment (pyphilia) [inline src/test/admin-tools.test.tsx:74]: Maybe give it a better name.

File: src/test/utils/utils.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import path from 'path';
+import Papa from 'papaparse';
+import { EXPORT_CSV_HEADERS } from '../../constants';
+
+const validateCsvData = (data: unknown[], numMessages: number): void => {
+  console.log(data);
```
Comment (pyphilia) [inline src/test/utils/utils.ts:6]: Remove.

**Review by pyphilia — CHANGES_REQUESTED** [review]
🐑

**Comment by spaenleh** [comment]
Here is an updated screenshot of the dialog that opens before clearing the chat, with an added button to save the chat:
Screenshot 2022-04-22 at 18 24 33

File: src/components/ClearChat.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        open={openConfirmation}
+        title={t('Clear Chat Confirmation')}
+        content={
+          <Box display="flex" flexDirection="column" alignItems="center">
+            <Typography>
+              {t(
+                'Do you want to clear all messages from this chat? This action is non-reversible. Use the button below if you want to make a backup.',
+              )}
+            </Typography>
+            <ExportChat variant="button" text="Save Chat" />
```
Comment (pyphilia) [inline src/components/ClearChat.tsx:39]: Translate save chat?

**Review by pyphilia — APPROVED** [review]
🦒

**Pull request merged**